python/graphframes/graphframe.py

```diff
@@ -102,6 +111,102 @@ def bfs(self, fromExpr, toExpr, edgeFilter=None, maxPathLength=10):
         jdf = builder.run()
         return DataFrame(jdf, self._sqlContext)
 
+    # Standard algorithms
+
+    def connected_components(self):
+        """
+        Computes the connected components of the graph.
+        :return:
+        """
+        jgf = self._jvm_graph.connectedComponents().run()
+        return _from_java_gf(jgf, self._sqlContext)
+
+    def label_propagation(self, max_steps):
+        """
+        Runs static label propagation for detecting communities in networks.
+        :param max_steps: the number of super steps to be performed.
+        :return:
+        """
+        jgf = self._jvm_graph.labelPropagation().setMaxSteps(max_steps).run()
+        return _from_java_gf(jgf, self._sqlContext)
+
+    def page_rank(self,
+                  reset_prob, source_id = None,
+                  fixed_num_iter = None,
+                  tolerance = None):
+        """
+        Runs the PageRank algorithm on the graph.
+        :param reset_prob:
+        :param source_id: (optional) the source vertex for a personalized PageRank.
+        :param fixed_num_iter: if set, the algorithm is run for a fixed number of iterations. In this case, the
+               `tolerance` parameter is ignored.
+        :param tolerance: if set, the algorithm is run until the given tolerance. In this case, the `fixed_num_iter`
+               parameter is ignored.
+        :return:
+        """
+        builder = self._jvm_graph.pageRank().setResetProbability(reset_prob)
+        if source_id is not None:
+            builder = builder.setSourceId(source_id)
+        if fixed_num_iter is not None:
+            builder = builder.fixedIterations(fixed_num_iter)
+        if tolerance is not None:
+            assert fixed_num_iter is None
+            builder = builder.untilConvergence(tolerance)
+        jgf = builder.run()
+        return _from_java_gf(jgf, self._sqlContext)
+
+    def shortest_paths(self, landmark_ids):
+        """
+        Runs the shortest path algorithm from a set of landmark vertices in the graph.
+        :param landmark_ids: a set of landmarks
+        :return:
+        """
+        jgf = self._jvm_graph.shortestPaths().setLandmarks(landmark_ids).run()
+        return _from_java_gf(jgf, self._sqlContext)
+
+    def strongly_connected_components(self, num_iterations):
+        """
+        Runs the strongly connected components algorithm on this graph.
+        :param num_iterations: the number of iterations to run.
+        :return:
+        """
+        jgf = self._jvm_graph.stronglyConnectedComponents().setNumIterations(num_iterations).run()
+        return _from_java_gf(jgf, self._sqlContext)
+
+    def svd_plus_plus(self, conf_dict):
+        """
+        Runs the SVD++ algorithm.
+        :param conf:
```

Comment:
* Scala: We should use a Builder which takes each parameter separately, not using a Conf.
* Python: We should use named arguments for each parameter separately, not use a Conf.